sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -72,13 +72,22 @@ static void boot_data_digest_compute(const void *boot_data,
  * @return Whether the entry is empty.
  */
 static hardened_bool_t boot_data_is_empty(const void *boot_data) {
-  for (size_t i = 0; i < kBootDataNumWords; ++i) {
-    if (read_32(boot_data) != kBootDataEmptyWordValue) {
-      return kHardenedBoolFalse;
-    }
+  static_assert(kBootDataEmptyWordValue == UINT32_MAX,
+                "kBootDataEmptyWordValue must be UINT32_MAX");
+  size_t i = 0;
+  hardened_bool_t is_empty = kHardenedBoolTrue;
+  uint32_t res = kBootDataEmptyWordValue;
+  for (; launder32(i) < kBootDataNumWords; ++i) {
+    res &= read_32(boot_data);
+    is_empty &= read_32(boot_data);
     boot_data = (char *)boot_data + sizeof(uint32_t);
   }
-  return kHardenedBoolTrue;
+  HARDENED_CHECK_EQ(i, kBootDataNumWords);
+  if (launder32(res) == kBootDataEmptyWordValue) {
+    HARDENED_CHECK_EQ(res, kBootDataEmptyWordValue);
+    return is_empty;
```

Comment:
Do we want to also HARDENED_CHECK_EQ `is_empty`? I would think that it correlates with `res`.

Reply:
Good point, I initially skimped on this to minimize overhead, which turned out to be negligible, but decided not to add it since there are hardened checks for the return value in `boot_data_page_info_update_impl()` (for both `kHardenedBoolTrue` and `kHardenedBoolFalse`).